### PR TITLE
Add support for rushjs.io monorepo for local builds (PR from another branch)

### PR DESCRIPTION
**Pull request by jpiyali**
New code structure:

1. root/vNext folder: root of configuration.
2. root/vNext/shared is for shared common packages.
3. root/vNext/extensions has all extensions include analytics.
4. Two SKUs are available under root/AISKU and root/AISKULight

Updated numerous package.json to include peer dependency for rollup (requirement), using npm version 4.5.x since this version is stable with rush build process.

Changes also include travis.yml changes to build beta packages through rush. Once checked in, I'll update the VSTS build definitions to the new paths. As a next step, i'll publish a round of beta packages and test E2E to ensure no breaking changes. Then will release beta 2.0.0 with pending breaking changes as discussed with Hector for customProperties, update core ITelemetryItem.ctx -> ITelemetryItem.ext and a few other minor changes.

### CODE REVIEW

File: vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts [hunk 1]
```diff
@@ -356,8 +356,8 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
      */
     public trackException(exception: IExceptionTelemetry, customProperties?: {[key: string]: any}): void {
         try {
-            let baseData = new Exception(this._logger, exception.exception, exception.properties, exception.measurements, exception.severityLevel)
-            let telemetryItem: ITelemetryItem = TelemetryItemCreator.create<IExceptionTelemetry>(
+            let baseData = new Exception(this._logger, exception.error, exception.properties, exception.measurements, exception.severityLevel)
```
Comment (jpiyali) [inline vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: this overall change is to fix build error

File: vNext/AISKU/package.json [hunk 2]
```diff
@@ -43,6 +47,7 @@
         "@microsoft/applicationinsights-properties-js": "1.0.0-beta.11"
     },
     "peerDependencies": {
-        "tslib": "^1.9.3"
+        "tslib": "^1.9.3",
+        "rollup": "^0.66.0"
```
Comment (markwolff) [inline vNext/AISKU/package.json]: This shouldn't be a peer dependecy, otherwise everyone will see an error when they try to install. I had problems with this being a devDep, so I just installed it to package.json during build then did `git checkout -- package.json` to remove it from deps after I finished building

File: vNext/AISKULight/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        "typescript": "2.5.3"
+    },
+    "dependencies": {
+        "@microsoft/applicationinsights-common": "1.0.0-beta.16",
+        "@microsoft/applicationinsights-channel-js": "1.0.0-beta.14",
+        "@microsoft/applicationinsights-core-js": "1.0.0-beta.5"
+    },
+    "peerDependencies": {
+      "tslib": "^1.9.3",
+      "rollup": "^0.66.0"
```
Comment (markwolff) [inline vNext/AISKULight/package.json]: same as rollup comment above

File: vNext/common/config/rush/command-line.json [hunk 4]
```diff
@@ -0,0 +1,189 @@
+/**
+ * This configuration file defines custom commands for the "rush" command-line.
+ * For full documentation, please see https://rushjs.io
+ */
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/command-line.schema.json",
+
+  /**
```
Comment (markwolff) [inline vNext/common/config/rush/command-line.json]: Can the comments be deleted? Is this file needed?

File: vNext/extensions/applicationinsights-analytics-js/package.json [hunk 5]
```diff
@@ -20,12 +20,19 @@
...
-        "@microsoft/applicationinsights-core-js": "^1.0.0-beta.5",
-        "@microsoft/applicationinsights-common": "^1.0.0-beta.16",
+        "@microsoft/applicationinsights-core-js": "1.0.0-beta.5",
+        "@microsoft/applicationinsights-common": "1.0.0-beta.16",
         "tslib": "^1.9.3"
     },
-    "license": "MIT"
+    "license": "MIT",
+    "peerDependencies": {
+        "rollup": "^0.66.0"
```
Comment (markwolff) [inline vNext/extensions/applicationinsights-analytics-js/package.json]: rollup comment

File: vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts [hunk 6]
```diff
@@ -356,8 +356,8 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
...
+            let telemetryItem: ITelemetryItem = TelemetryItemCreator.create<Exception>(
```
Comment (markwolff) [inline vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: 👍

File: vNext/extensions/applicationinsights-properties-js/package.json [hunk 7]
```diff
@@ -27,9 +27,12 @@
...
-        "@microsoft/applicationinsights-core-js": "^1.0.0-beta.5",
-        "@microsoft/applicationinsights-common": "^1.0.0-beta.16",
+        "@microsoft/applicationinsights-core-js": "1.0.0-beta.5",
+        "@microsoft/applicationinsights-common": "1.0.0-beta.16",
         "tslib": "^1.9.3"
     },
-    "license": "MIT"
+    "license": "MIT",
+    "peerDependencies": {
+        "rollup": "^0.66.0"
```
Comment (markwolff) [inline vNext/extensions/applicationinsights-properties-js/package.json]: rollup comment

File: vNext/shared/AppInsightsCommon/package.json [hunk 8]
```diff
@@ -25,11 +25,15 @@
...
     },
     "dependencies": {
-        "@microsoft/applicationinsights-core-js": "^1.0.0-beta.5",
+        "@microsoft/applicationinsights-core-js": "1.0.0-beta.5",
         "tslib": "^1.9.3"
     },
-    "license": "MIT"
+    "license": "MIT",
+    "peerDependencies": {
+        "rollup": "^0.66.0"
```
Comment (markwolff) [inline vNext/shared/AppInsightsCommon/package.json]: rollup comment

File: .travis.yml [hunk 9]
```diff
@@ -1,5 +1,42 @@
 language: node_js
 node_js:
-  - "10.9"
+  - "8.9.0"
+before_install:
+  - npm i -g npm@4.5.0
```
[the remark this answers is not in the document]
Comment (piyalijana) [reply .travis.yml]: I can try the original value again, but IIRC, there were integrity check issues with rush and that version.

File: vNext/extensions/applicationinsights-properties-js/package.json (hunk 7 above)
Comment (jpiyali) [reply vNext/extensions/applicationinsights-properties-js/package.json]: Moved to devDependencies

File: vNext/shared/AppInsightsCommon/package.json (hunk 8 above)
Comment (jpiyali) [reply vNext/shared/AppInsightsCommon/package.json]: Moved to devDependencies

File: vNext/extensions/applicationinsights-analytics-js/package.json (hunk 5 above)
Comment (jpiyali) [reply vNext/extensions/applicationinsights-analytics-js/package.json]: Moved to devDependencies

File: vNext/AISKULight/package.json (hunk 3 above)
Comment (jpiyali) [reply vNext/AISKULight/package.json]: Moved to devDependencies

File: vNext/AISKU/package.json (hunk 2 above)
Comment (jpiyali) [reply vNext/AISKU/package.json]: Moved to devDependencies. Build worked.

File: .travis.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (jpiyali) [reply .travis.yml]: reverted to 10.9. Works.

File: vNext/common/config/rush/command-line.json (hunk 4 above)
Comment (jpiyali) [reply vNext/common/config/rush/command-line.json]: Cleaned up comments. added support for test command.
